pkg/collect/util.go

```diff
@@ -227,3 +228,21 @@ func getTLSParamsFromSecret(ctx context.Context, client kubernetes.Interface, se
 
 	return caCert, clientCert, clientKey, nil
 }
+
+func checkForExistingServiceAccount(client kubernetes.Interface, namespace string, serviceAccountName string) error {
+	_, err := client.CoreV1().ServiceAccounts(namespace).Get(context.Background(), serviceAccountName, metav1.GetOptions{})
+
+	if err != nil {
+		klog.V(2).Infof("Failed to get service account %s, creating it", serviceAccountName)
+		// Service account doesn't exist, create it
+		_, err = client.CoreV1().ServiceAccounts(namespace).Create(context.Background(), &corev1.ServiceAccount{
```

Comment:
this will help immensely in tests with k3s, but we want to be sure that it's the behavior we want with production sites - the end user might not expect us to be creating serviceAccount resources when running Troubleshoot, and everything else gets deleted at the end.  I'd not want to go deleting the default sa in this case.

Reply:
I have changed the logic that we just get if the service account existed or not. Not creating it